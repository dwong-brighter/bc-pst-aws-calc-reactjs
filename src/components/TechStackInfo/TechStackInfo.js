import React from 'react';
import { makeStyles } from '@material-ui/styles';
import Avatar from '@material-ui/core/Avatar';
import ExpandMoreIcon from '@material-ui/icons/ExpandMoreRounded';
import ExpansionPanel from '@material-ui/core/ExpansionPanel';
import ExpansionPanelDetails from '@material-ui/core/ExpansionPanelDetails';
import ExpansionPanelSummary from '@material-ui/core/ExpansionPanelSummary';
import List from '@material-ui/core/List';
import ListItem from '@material-ui/core/ListItem';
import ListItemText from '@material-ui/core/ListItemText';
import logo from '../../logo.svg';
import Typography from '@material-ui/core/Typography';

const useStyles = makeStyles({
  '@keyframes logoSpin': {
    from: { transform: 'rotate(0deg)' },
    to: { transform: 'rotate(360deg)' },
  },
  logo: {
    animation: '$logoSpin infinite 20s linear',
    marginLeft: '0.5rem',
    marginTop: '0.5rem',
  },
  details: {
    width: '100%',
    display: 'block',
  },
});

function TechStackInfo() {
  const classes = useStyles();

  return (
    <ExpansionPanel defaultExpanded>
      <ExpansionPanelSummary expandIcon={<ExpandMoreIcon aria-label="Expand" />}>
        <Typography variant="subtitle1" color="inherit">Technology Stack</Typography>
      </ExpansionPanelSummary>
      <ExpansionPanelDetails className={classes.details}>
        <Avatar className={classes.logo} src={logo} alt="logo" />
        <List>
          <ListItem dense>
            <ListItemText primary="ReactJS" />
          </ListItem>
          <ListItem dense>
            <ListItemText primary="Material-UI" />
          </ListItem>
          <ListItem dense>
            <ListItemText primary="Node.js (and NPM)" />
          </ListItem>
        </List>
      </ExpansionPanelDetails>
    </ExpansionPanel>
  );
}

TechStackInfo.propTypes = {};

export default TechStackInfo;
