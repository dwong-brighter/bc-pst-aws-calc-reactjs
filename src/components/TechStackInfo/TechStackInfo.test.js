import React from 'react';
import { shallow } from 'enzyme';
import TechStackInfo from './TechStackInfo';

describe('<TechStackInfo />', () => {
  it('renders without crashing', () => {
    const wrapper = shallow(<TechStackInfo />);
    expect(wrapper).toMatchSnapshot();
  });
});
