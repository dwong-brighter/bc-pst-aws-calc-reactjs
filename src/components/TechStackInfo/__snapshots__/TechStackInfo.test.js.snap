// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TechStackInfo /> renders 1`] = `
<WithStyles(ForwardRef(ExpansionPanel))
  defaultExpanded={true}
>
  <WithStyles(ForwardRef(ExpansionPanelSummary))
    expandIcon={
      <Memo
        aria-label="Expand"
      />
    }
  >
    <WithStyles(ForwardRef(Typography))
      color="inherit"
      variant="subtitle1"
    >
      Technology Stack
    </WithStyles(ForwardRef(Typography))>
  </WithStyles(ForwardRef(ExpansionPanelSummary))>
  <WithStyles(ForwardRef(ExpansionPanelDetails))
    className="makeStyles-details-3"
  >
    <WithStyles(ForwardRef(Avatar))
      alt="logo"
      className="makeStyles-logo-2"
      src="logo.svg"
    />
    <WithStyles(ForwardRef(List))>
      <WithStyles(ForwardRef(ListItem))
        dense={true}
      >
        <WithStyles(ForwardRef(ListItemText))
          primary="ReactJS"
        />
      </WithStyles(ForwardRef(ListItem))>
      <WithStyles(ForwardRef(ListItem))
        dense={true}
      >
        <WithStyles(ForwardRef(ListItemText))
          primary="Material-UI"
        />
      </WithStyles(ForwardRef(ListItem))>
      <WithStyles(ForwardRef(ListItem))
        dense={true}
      >
        <WithStyles(ForwardRef(ListItemText))
          primary="Node.js (and NPM)"
        />
      </WithStyles(ForwardRef(ListItem))>
    </WithStyles(ForwardRef(List))>
  </WithStyles(ForwardRef(ExpansionPanelDetails))>
</WithStyles(ForwardRef(ExpansionPanel))>
`;
